plugins/calibrecompanion.koplugin/main.lua

```diff
@@ -315,11 +315,52 @@ function CalibreCompanion:sendBooklists(arg)
     DEBUG("SEND_BOOKLISTS", arg)
 end
 
+-- TODO: move to more central location
+local lfs = require("libs/libkoreader-lfs")
+
+-- We make the simplifying assumption that no-one will want to store books with
+-- a literal \ in the filename.
+-- FIXME: can calibre-on-windows even send paths with \ in them? If not we don't
+-- need to worry about this.
+local function normalize_path(path)
+    return (path:gsub('\\', '/'):gsub('/+', '/'))
+end
+
+local function path_exists(path)
+    return lfs.attributes(path, "mode") ~= nil
+end
+
+-- strip the trailing element from path
+-- note: dirname foo/bar and dirname foo/bar/ are both foo
+-- dirname / is /
+-- dirname foo is .
+local function dirname(path)
```

Comment:
Ah, it's too bad you wrote this. We have some util functions for that.

http://koreader.rocks/doc/modules/util.html#splitFilePathName

Reply:
No worries, I had these functions lying around already, so it wasn't like I put a lot of time into it. :)